loop.go

```diff
@@ -0,0 +1,17 @@
+package main
+
+import (
+	"time"
+)
+
+func FetchLoop(state *State, interval int) {
+	for {
+		select {
+		case <-threadDone:
```

Comment:
Where does `threadDone` come from? Global?

Reply:
It is global, yes, defined in `main.go`, would you do it differently?